service/history/workflow/update/update.go

```diff
@@ -87,9 +87,27 @@ type (
 	// (OnMessage calls) must be done while holding the workflow lock.
 	Update struct {
 		// accessed only while holding workflow lock
-		id              string
-		state           state
-		request         *anypb.Any // of type *updatepb.Request, nil when not in stateAdmitted or stateSent.
+		id    string
+		state state
+		// request is nil when:
+		//   - update is in stateAccepted,
+		//   - update was restored in stateAdmitted with newAdmitted from UpdateInfo.AdmissionInfo.
+		//       We don't load request form event because if there is durable UpdateAdmitted event with request,
+		//       then we don't write this request 2nd time to UpdateAccepted event.
+
```

Comment:
This might have been my bad with the GitHub UI, but I think we shoudl delete the original comment.

```suggestion
```

Reply:
My bad :-(